includes/functions.php

```diff
@@ -1488,3 +1491,51 @@ function describe_bgp_error_code($code, $subcode)
 
     return $message;
 }
+
+/**
+ * Gets stored app data JSON and returns it.
+ * If none is present or it is bad, a blank array is returned.
+ *
+ * @params int $app_id
+ *
+ * @return array
+ */
+function get_app_data($app_id)
```

Comment:
Question why a two whole functions instead of just using
`Application::find($app_id)->data`

That will return a decoded array (and also encode the array when saved)  if  you set `protected $casts = ['data' => 'array'];` on the Application model.

This also has the added benefit of caching the data if you pass around the loaded Application object.

Reply:
Three whole functions, I am a bit lost, I seem to be missing the third.

As to why, it seemed it would be easier than calling something like this each time.

```
$app = Application::where(['app_id' => $app_id])->first();
$parsed_json = json_decode($app->data, true);
if (json_last_error() !== JSON_ERROR_NONE) {
    //whatever
}
```

That said if you know how to get `app/Models/Application.php` to play nice with a helper function or the like for this, that would also be cool and I am totally open to any input on actually getting that to work.

Reply:
So a fresh set of eyes on it and got a save helper function working...

```
   public function save_data($data = [])
   {
      $this->fill(['data'=>json_encode($data)]);
      $this->save();
   }
```

Aye. Going to be updating this and ditching the functions in `include/functions.php`.

Reply:
Woot! All replaced with helper functions in the model now every where. Docs updated as well.